extensions/amp-story/1.0/bookend/bookend-component.js

```diff
@@ -30,14 +30,29 @@ export let BookendDataDef;
  */
 export let BookendComponentDef;
 
+const articleComponent = new ArticleComponent();
+const articleTitleComponent = new ArticleTitleComponent();
+
+/**
+ * @typedef {(!ArticleComponent|!ArticleTitleComponent)}
+ */
+export let BookendComponentClass;
+
 /**
- * Map used to dispatch the components to their specific builder classes.
- * @const {!Object<string, ./components/abstract.AbstractBookendComponent>}
+ * Dispatches the components to their specific builder classes.
+ * @param {string} componentType
+ * @return {?BookendComponentClass}
  */
-export const componentsMap = {
-  'small': ArticleComponent,
-  'article-set-title': ArticleTitle,
-};
+function componentClassFor(componentType) {
+  switch (componentType) {
+    case 'small':
+      return articleComponent;
```

Comment:
This code is very confusing for two reasons:

- the function is named ``componentClassFor`` when it actually returns an instance, and not a class
- the ``articleComponent`` name really makes it feel like it's going to be an instance of a component, where one could expect to find properties (the component data) etc, when it's actually a singleton that behaves as a builder

What do you think of renaming:

- ``componentClassFor`` into ``componentBuiderInstanceFor``
- ``articleComponent`` into ``articleComponentBuilder``